cassandra/query.py

```diff
@@ -701,6 +720,16 @@ def _maybe_set_routing_attributes(self, statement):
                 self.routing_key = statement.routing_key
                 self.keyspace = statement.keyspace
 
+    def _update_custom_payload(self, statement):
+        if statement.custom_payload:
+            if self.custom_payload is None:
+                self.custom_payload = {}
+            self.custom_payload.update(statement.custom_payload)
```

Comment:
My thinking was that statements would only have payloads on them if they were explicitly set. And if they're being added to the batch, they want those values (the sub-statements don't get to carry individual payloads).

This felt like the most flexible and easy to use. I don't feel too strongly about it since I doubt it will ever be used,